File: tests/libjournal/channel_inventory.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2020 all rights reserved


// get the journal
#include <j2/journal.h>
// support
#include <cassert>


// severity stub
class severity {};
// channel realization
using channel_t = pyre::journal::channel_t<severity, pyre::journal::inventory_t<true>>;


// exercise the channel state interface
int main() {
    // make a channel
    channel_t channel("test.channel");

    // get its inventory
    auto & inventory = channel.inventory();
    // verify it is on, by default
    assert (inventory.state());
    // and that its device is null
    assert (!inventory.device());

    // all done
    return 0;
}


// end of file

File: tests/journal.pkg/channel_inventory.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2020 all rights reserved


# externals
import collections  # for {defaultdict}


# verify that the default channel state is as expected
def test():
    """
    Exercise the default channel state
    """
    # get the package
    from j2.Channel import Channel
    # derive a severity
    class Severity(Channel):
        # make and attach an index
        index = collections.defaultdict(Channel.disabled_type)

    # make one
    channel = Severity(name="test.channel")

    # get its inventory
    inventory = channel.inventory
    # verify that it is disabled
    assert inventory.state == False
    # and that it has no registered device
    assert inventory.device == None

    # all done
    return


# main
if __name__ == "__main__":
    # run the test
    test()


# end of file
